src/main/kotlin/com/terraformation/backend/documentproducer/DocumentUpgradeCalculator.kt

```diff
@@ -190,18 +185,7 @@ class DocumentUpgradeCalculator(
             }
           }
 
-      val deleteOperations =
-          sectionValues.mapNotNull { sectionValue ->
-            if (sectionValue.value is SectionValueVariable &&
-                sectionValue.value.usedVariableId !in newManifestVariables &&
-                sectionValue.value.usedVariableId !in replacementVariableIds) {
-              DeleteValueOperation(projectId, sectionValue.id)
-            } else {
-              null
-            }
-          }
-
-      updateOperations + deleteOperations
+      updateOperations
```

Comment:
I guess this variable can go away and the `mapNotNull` call can be the value of the `if` block.